### PR TITLE
Dc 2982 frontend changes

### PR DESCRIPTION
**Pull request by PriyaBasker23**
Here is a checklist of design-related tasks to finalise on the prototype, aligning it with the specifications outlined in
https://www.figma.com/file/zTOVhHRnWfxRJqH6Gr8cIc/Catalogue-sketches?type=design&node-id=206-2398&mode=design&t=pEjdb6k2pwuuHuiw-0

completed the following task as per the following list 
- [x]  Ensure links have the govuk-link class.
- [x] Modify filters to include the desired options when initially presented to users (e.g., domain, data product).
- [x] Integrate pagination buttons at the bottom of the page.
- [x]  Remove the horizontal rule between search results or eliminate lines from the summary list component.
- [x] Incorporate the details page (accessible by clicking on a result).
- [x]  Apply Filter
- [x] Clear Filter
- [x] Remove Filters
- [x] Back buttons
- [x] Basic details page template

### CODE REVIEW

File: static/assets/js/htmx.min.js
Comment (mitchdawson1982) [inline static/assets/js/htmx.min.js:1]: We are loading htmx from CDN in ./base/head.html and adding this file into the static directory. We should load it from one place only.

File: home/urls.py [hunk 1]
```diff
@@ -7,4 +7,5 @@
 urlpatterns = [
     path("", views.home_view, name="home"),
     path("search", views.search_view, name="search"),
+    path("details", views.details_view, name="details"),
```
Comment (mitchdawson1982) [inline home/urls.py:10]: We should refactor the details url pattern to accept an id/pk value like this. 
`path(details/<str:id>/` . 
We can then build the individual url in the search results likes this. `"{% url 'home:details' id=result.id %}"`

File: home/views.py [hunk 2]
```diff
@@ -1,25 +1,99 @@
...
     client = get_catalogue_client()
-    search_results = client.search(query=query, page=page)
 
+    # Fetch domainlist without query
+    search_results = client.search(query="", page=None)
     context = {}
+    domain_list = search_results.facets['domains']
+    context["domainlist"] = domain_list
+
+    if request.GET.getlist("domain"):
```
Comment (mitchdawson1982) [inline home/views.py:46]:
```suggestion
  domains = request.GET.getlist("domain", [])
  if domains:
      selected_domain = filter_seleted_domains(domain_list, domains)
      context['selected_domain'] = selected_domain
      request.session['selected_domain'] = selected_domain
      request.session['domains'] = domains
      filter_value = [MultiSelectFilter("domains", domains)]
```

File: home/views.py [hunk 3]
```diff
@@ -1,25 +1,99 @@
...
+
+        # Populated selected domain
+        selected_domain = filter_seleted_domains(domain_list, domains)
+        context['domains'] = domains
+        context['selected_domain'] = selected_domain
+
+        # Reassign to session
+        request.session['selected_domain'] = selected_domain
+        request.session['domains'] = domains
+        if not domains:
```
Comment (mitchdawson1982) [inline home/views.py:73]: Could this check be performed further up before session assignment?

File: pyproject.toml [hunk 4]
```diff
@@ -9,6 +9,7 @@ readme = "README.md"
 python = "^3.11"
 django = "^5.0.1"
 pyyaml = "^6.0.1"
+htmx = "^0.0.0"
```
Comment (mitchdawson1982) [inline pyproject.toml:12]: Can this be removed ?

File: templates/base/head.html [hunk 5]
```diff
@@ -11,5 +11,6 @@
   <link rel="manifest" href="{% static 'assets/manifest.json' %}">
   <link rel="stylesheet" type="text/css" href="{% static 'assets/css/search.css' %}">
   <link rel="stylesheet" type="text/css" href="{% static 'assets/css/base.css' %}">
+  <script src="https://unpkg.com/htmx.org@1.7.0/dist/htmx.js"></script>
```
Comment (mitchdawson1982) [inline templates/base/head.html:14]: Can htmx be removed ?

File: templates/partial/search_result.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{% load markdown %}
+{% for result in results %}
+
+<div class="govuk-grid-row" id="search_result">
+    <div class="govuk-grid-column-full">
+      <h3 class="govuk-heading-m govuk-!-margin-bottom-2">
+          <a href="{% url 'home:details' %}?id={{result.id}}" class="govuk-link">{{result.name}}</a>
```
Comment (mitchdawson1982) [inline templates/partial/search_result.html:7]: We should refactor the url generation to pass the id into the function automatically.

File: pyproject.toml (hunk 4 above)
Comment (PriyaBasker23) [reply pyproject.toml:12]: yes

File: home/views.py (hunk 3 above)
Comment (PriyaBasker23) [reply home/views.py:79]: We need to reset the session with latest information . So we cant do that

**Review by mitchdawson1982 — APPROVED** [review]
LGTM

**Pull request merged**